javascript-loops-practice/filter-out-strings.js

```diff
@@ -0,0 +1,10 @@
+/* exported filterOutStrings */
+function filterOutStrings(strings) {
+  var noStrings = [];
+  for (var i = 0; i < strings.length; i++) {
+    if (typeof (strings[i]) === 'string') {
+      delete strings[i];
```

Comment:
Oh noooooo, this is bad news. You should **_never_** change the original value in an argument like this. The only work here that actually matters is in your `else` block, you should just make that your primary condition. Instead of `typeof (strings[i]) === 'string'` for your condition, you can do `typeof (strings[I]) !== 'string' in order to make your `else` block your main `if` block